app/app.go

```diff
@@ -758,9 +758,9 @@ func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino
 	paramsKeeper.Subspace(distrtypes.ModuleName)
 	paramsKeeper.Subspace(slashingtypes.ModuleName)
 	paramsKeeper.Subspace(govtypes.ModuleName)
-	paramsKeeper.Subspace(ibctransfertypes.ModuleName)
+	paramsKeeper.Subspace(ibctransfertypes.ModuleName).WithKeyTable(ibctransfertypes.ParamKeyTable())
 	paramsKeeper.Subspace(ibcexported.ModuleName)
-	paramsKeeper.Subspace(icahosttypes.SubModuleName)
+	paramsKeeper.Subspace(icahosttypes.SubModuleName).WithKeyTable(icahosttypes.ParamKeyTable())
```

Comment:
[no change needed] I wonder how this interacts with https://github.com/celestiaorg/celestia-app/pull/5442

For context: I noticed that the ICA host params on v6 defaulted to the upstream ICA host params (a.k.a wildcard for AllowMessages) and I had to explicitly set them in the v5 -> v6 upgrade handler. But I wonder if that migration is not needed after this `.WithKeyTable` change.

Do you know why the icahosttypes and ibctransfertypes param subspaces need the `.WithKeyTable` lines but the other subspaces don't need them?

Reply:
I think your change will still be required, the WithKeyTable change makes it so that the icahost module doesn't fail silenly (fails to claim capability)

I tried running

```
$ celestia-appd query params subspace icahost AllowMessages
param:
  key: AllowMessages
  subspace: icahost
``` 

It looks like here we have the key present, but it has no value, so I would imagine it defaults to the `*`

However we should double check that both changes work together 👀 maybe we can extend the test to query the allow messages.